jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -377,19 +377,22 @@ protected void sendWelcome(HttpContent content, String pathInContext, boolean en
         if (!endsWithSlash)
         {
             StringBuffer buf = request.getRequestURL();
-            int param = buf.lastIndexOf(";");
-            if (param < 0)
-                buf.append('/');
-            else
-                buf.insert(param, '/');
-            String q = request.getQueryString();
-            if (q != null && q.length() != 0)
+            synchronized (buf)
```

Comment:
This looks odd.
It looks like we are attempting to modify an immutable field in the request object?
That can't be right.

Reply:
:rofl: @gregw